pyxem/generators/variance_generator.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Variance generators in real and reciprocal space for fluctuation electron
+microscopy.
+
+"""
+
+import numpy as np
+from hyperspy.signals import Signal2D
+from hyperspy.api import stack
+
+from pyxem.signals.diffraction_variance import DiffractionVariance
+from pyxem.signals.diffraction_variance import ImageVariance
+
+
+class VarianceGenerator():
+    """Generates variance images for a specified signal and set of aperture
+    positions.
+
+    Parameters
+    ----------
+    signal : ElectronDiffraction2D
+        The signal of electron diffraction patterns to be indexed.
+
+    """
+
+    def __init__(self, signal, *args, **kwargs):
+        self.signal = signal
+
+        # add a check for calibration
+
+    def get_diffraction_variance(self, dqe):
+        """Calculates the variance in scattered intensity as a function of
```

Comment:
Need to say something explicitly here about the Poisson noise scaling thing.

Reply:
This was marked resolved without any meaningful change to the code - so reopening. I mean it needs a reference and to write the equation out in the docstring.

Reply:
Yeah whoops, accidentally closed the wrong one it seems.